src/aliases/api/handlers/handlers.go

```diff
@@ -0,0 +1,299 @@
+// @Router /aliases/{registry_name}/aliases/{alias_key} [put]
+func (h *AliasHandler) updateAlias(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	// should always exists in this subrouter
+	regName := vars["registry_name"]
+	key := vars["alias_key"]
+
+	err := validatePathVars(regName, key)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	var aliasReq types.AliasRequest
+	err = jsonutils.UnmarshalBody(r.Body, &aliasReq)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	alias := &aliasent.Alias{
+		RegistryName: aliasent.RegistryName(regName),
+		Key:          aliasent.AliasKey(key),
+		Value:        aliasent.AliasValue(aliasReq.Value),
+	}
+	// TODO the: we have to either:
+	// - use an ID PK in the alias table to be able to update the alias key while renaming the alias
+	// - modify the UpdateAlias func to change the alias key (PK)
+	// - delete + create of the new alias
+	alias, err = h.alias.UpdateAlias(r.Context(), aliasent.RegistryName(regName), *alias)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+
+	err = jsonWrite(w, types.AliasResponse{
+		Value: types.AliasValue(alias.Value),
+	})
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+}
+
+// deleteAlias deletes an alias value.
+// @Summary Delete an alias
+// @Description Delete an alias of a key from a dedicated alias registry
+// @Tags alias
+// @Param registry_name path string true "registry identifier"
+// @Param alias_key path string true "alias identifier"
+// @Param request body types.DeleteAliasRequest true "Delete Alias Request"
+// @Success 204 "Deleted successfully"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
+// @Failure 500 {object} ErrorResponse "Internal server error"
+// @Router /aliases/{registry_name}/aliases/{alias_key} [delete]
+func (h *AliasHandler) deleteAlias(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	// should always exists in this subrouter
+	regName := vars["registry_name"]
+	key := vars["alias_key"]
+
+	err := validatePathVars(regName, key)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	err = h.alias.DeleteAlias(r.Context(), aliasent.RegistryName(regName), aliasent.AliasKey(key))
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// @Summary Get all the aliases in a registry
+// @Description Get all the aliases in a registry
+// @Tags alias
+// @Produce json
+// @Param registry_name path string true "registry identifier"
+// @Param alias_key path string true "alias identifier"
+// @Success 200 {array} types.GetAliasResponse "a list of Aliases"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
```

Comment:
is it really possible to send an invalid request format?

Reply:
Yes, in the case the registry_name in the path doesn't respect the format
```go
const pathVarsFormat = "^[a-zA-Z0-9-_+]+$"
```

Reply:
but in case path does not match I would expect a 404 `Page Not Found` instead of 400. isn't that right?

Reply:
From https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400
> 400 Bad Request
> 
> The HyperText Transfer Protocol (HTTP) 400 Bad Request response status code indicates that the server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).

I would consider this a `deceptive request routing`.
Given their browser, they maybe want an alias with a space ` `, and the browser could display the url as `/registries/my-registry/aliases/my%20alias`. That could confuse them.

Reply:
I also get confused by the fact a ID are defined in URl and then you can get a 400 but it is fine I would say